public/docs/dart/latest/tutorial/toh-pt3.jade

```diff
@@ -1,14 +1,251 @@
+  At the moment, the *Heroes* and *Hero Detail* views are combined in one template in `AppComponent`.
+  Let’s **cut** the *Hero Detail* content from `AppComponent` and **paste** it into the new template property of  `HeroDetailComponent`.
+
+  We previously bound to the `selectedHero.name` property of the `AppComponent`.
+  Our `HeroDetailComponent` will have a `hero` property, not a `selectedHero` property.
+  So we replace `selectedHero` with `hero` everywhere in our new template. That's our only change.
+  The result looks like this:
+
++makeExample('toh-3/dart/lib/hero_detail_component.dart', 'template', 'hero_detail_component.dart (template)')(format=".")
+
+:marked
+  Now our hero detail layout exists only in the `HeroDetailComponent`.
+
+  #### Add the *hero* property
+  Let’s add that `hero` property we were talking about to the component class.
++makeExample('toh-3/dart/lib/hero_detail_component.dart', 'hero')(format=".")
+:marked
+  Uh oh. We declared the `hero` property as type `Hero` but our `Hero` class is over in the `app_component.dart` file.
+  We have two components, each in their own file, that need to reference the `Hero` class.
+
+  We solve the problem by relocating the `Hero` class from `app_component.dart` to its own `hero.dart` file.
+
++makeExample('toh-3/dart/lib/hero.dart', null, 'hero.dart (Exported Hero class)')(format=".")
+
+:marked
+  Add the following import statement near the top of both `app_component.dart` and `hero_detail_component.dart`.
+
++makeExample('toh-3/dart/lib/hero_detail_component.dart', 'hero-import', 'hero_detail_component.dart and app_component.dart (Import the Hero class)')(format=".")
+
+:marked
+  #### The *hero* property is an ***input***
+
+  The `HeroDetailComponent` must be told what hero to display. Who will tell it? The parent `AppComponent`!
+
+  The `AppComponent` knows which hero to show: the hero that the user selected from the list.
+  The user's selection is in its `selectedHero` property.
+
+  We will soon update the `AppComponent` template so that it binds its `selectedHero` property
+  to the `hero` property of our `HeroDetailComponent`. The binding *might* look like this:
+code-example(format=".").
+  &lt;my-hero-detail [hero]="selectedHero">&lt;/my-hero-detail>
+:marked
+  Notice that the `hero` property is the ***target*** of a property binding &mdash; it's in square brackets to the left of the (=).
+
+  Angular insists that we declare a ***target*** property to be an ***input*** property.
+  If we don't, Angular rejects the binding and throws an error.
+.l-sub-section
+  :marked
+    We explain input properties in more detail [here](../guide/attribute-directives.html#why-input)
+    where we also explain why *target* properties require this special treament and
+    *source* properties do not.
+:marked
+  There are a couple of ways we can declare that `hero` is an *input*.
+  We'll do it the way we *prefer*, by annotating the `hero` property with the `@Input` decorator.
```

Comment:
IMHO in Dart it's called annotation instead of decorator. TS calls it decorator. They also have annotations AFAIK but I can't tell when it's the one and when the other.